doc/concepts/configuration.rst

```diff
@@ -0,0 +1,651 @@
+    :language: lua
+    :dedent:
+
+As a result of :ref:`starting <configuration_run_instance>` the *instance001*, a log should contain the following line:
+
+.. code-block:: console
+
+    main/103/interactive/myapp I> Hello from app, instance001!
+
+The ``app`` section can be placed in any :ref:`configuration scope <configuration_scopes>`.
+As an example use case, you can provide different applications for storages and routers in a sharded cluster:
+
+.. code-block:: yaml
+
+    groups:
+      storages:
+        app:
+          module: storage
+          # ...
+      routers:
+        app:
+          module: router
+          # ...
+
+Learn more about using Tarantool as the application server from :ref:`Developing applications with Tarantool <how-to-app-server>`.
+
+
+
+.. _configuration_predefined_variables:
+
+Predefined variables
+********************
+
+In a configuration file, you can use the following predefined variables that are replaced with actual values at runtime:
+
+-   ``instance_name``
+-   ``replicaset_name``
+-   ``group_name``
+
+To reference such variables in a configuration file, use double curly braces.
+In the example below, ``{{ instance_name }}`` is replaced with *instance001*.
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/templating/config.yaml
+    :language: yaml
+    :dedent:
+
+As a result, the :ref:`paths to snapshots and write-ahead logs <configuration_options_directories>` differ for different instances.
+
+
+
+..  _configuration_environment_variable:
+
+Environment variables
+~~~~~~~~~~~~~~~~~~~~~
+
+For each configuration parameter, Tarantool provides two sets of predefined environment variables:
+
+*   Variables whose names start with ``TT_`` are used to substitute parameters specified in a configuration file.
+    This means that these variables have a higher :ref:`priority <configuration_precedence>` than the options specified in a configuration file.
+
+*   Variables whose names start with ``TT_`` and end with ``_DEFAULT`` are used to specify default values for parameters missing in a configuration file.
+    These variables have a lower :ref:`priority <configuration_precedence>` than the options specified in a configuration file.
+
+For example, ``TT_IPROTO_LISTEN`` and ``TT_IPROTO_LISTEN_DEFAULT`` correspond to the ``iproto.listen`` option.
+``TT_SNAPSHOT_DIR`` and ``TT_SNAPSHOT_DIR_DEFAULT`` correspond to the ``snapshot.dir`` option.
+To see all the supported environment variables, execute the ``tarantool`` command with the ``--help-env-list`` :ref:`option <configuration_command_options>`.
+
+.. code-block:: console
+
+    $ tarantool --help-env-list
+
+Below are a few examples that show how to set environment variables of different types, like *string*, *number*, *array*, or *map*:
+
+*   (*String*) In the example below, ``TT_IPROTO_LISTEN`` is used to specify a :ref:`listening host and port <configuration_options_connection>` values:
```

Comment:
Punctuation in this list is confusing me. I don't get these brackets and italic. Why not just
```
- String. In the example ...
- Number. In this example...
-  Array. ...
```